include/robot_interfaces/cereal_eigen.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file
+ * @brief Serialization functions for serializing Eigen matrices and arrays
+ *        with cereal.
+ * @authors Azoth, eudoxos
+ * @date 2020-01-15
+ * @license CC BY-SA 4.0
+ * @todo Move this to some "serialization tools" package.
+ *
+ * Taken from https://stackoverflow.com/a/51944389/2095383.
+ */
+#include <Eigen/Eigen>
+#include <type_traits>
```

Comment:
these functions are a little but obscure to me. one important question: is there any problem with memory allocation by using these functions? in particular, by using this stuff inside the timeseries, is any memory being allocated? that would be an issue.

Reply:
This is indeed a valid concern.  I searched for a while but did not find any information on this (I also looked into the code but did not understand enough of it).  Maybe @vincentberenz knows more about this?

Reply:
@luator @mwuethri What function are you mentionning ? The code snippet above shows some include

Reply:
@vincentberenz Sorry, I missed your question earlier.  At least I was referring to cereal in general which you basically answered below.